ggml-sycl.cpp

```diff
@@ -3325,7 +3328,7 @@ class sycl_gpu_mgr {
                 dpct::device_info prop;
                 dpct::get_device_info(prop, device);
                 if (max_compute_units == prop.get_max_compute_units() &&
-                    is_ext_oneapi_device(device)) {
+                    backend == device.get_backend()) {
```

Comment:
The more I think about it, this should be checking for the same SYCL platform, not the same SYCL backend.  If we check for the same SYCL backend we could still get into trouble trying to create an invalid SYCL context if two GPUs in different platforms happen to have the same number of compute units and the same backend.  This probably won't happen in practice, but it's easy enough to check the platform instead.